src/snowflake/snowpark/session.py

```diff
@@ -1623,8 +1623,14 @@ def query_history(self) -> QueryHistory:
         return query_listener
 
     def _table_exists(self, table_name: str):
-        tables = self._run_query(f"show tables like '{table_name}'")
-        return tables is not None and len(tables) > 0
+        try:
+            # DESC is used here because SHOW does not work for qualified table name
```

Comment:
add one more comment saying we use `cursor.describe` so the failed query won't show up in query history?